cadquery/cq.py

```diff
@@ -2728,6 +2728,66 @@ def _sweep(self, path, multisection=False, makeSolid=True, isFrenet=False,
 
         return Compound.makeCompound(toFuse)
 
+    def interpPlate(self, surf_edges, surf_pts=[], thickness=0, combine=True, clean=True, Degree=3, NbPtsOnCur=15, NbIter=2, Anisotropie=False, Tol2d=0.00001, Tol3d=0.0001, TolAng=0.01, TolCurv=0.1, MaxDeg=8, MaxSegments=9):
+        """
+        Returns a plate surface that is 'thickness' thick, enclosed by 'surf_edge_pts' points,  and going through 'surf_pts' points.
+        
+        :param surf_edges
+        :type 1 surf_edges: list of [x,y,z] float ordered coordinates
+        :type 2 surf_edges: list of unordered CadQuery wires
+        :param surf_pts = [] (uses only edges if [])
+        :type surf_pts: list of [x,y,z] float coordinates
+        :param thickness = 0 (returns 2D surface if 0)
+        :type thickness: float >= 0
+        :param combine: should the results be combined with other solids on the stack
+            (and each other)?
+        :type combine: true to combine shapes, false otherwise.
+        :param boolean clean: call :py:meth:`clean` afterwards to have a clean shape      
+        :param Degree = 3 (OCCT default)
+        :type Degree: Integer >= 2
+        :param NbPtsOnCur = 15 (OCCT default)
+        :type: NbPtsOnCur Integer >= 15
+        :param NbIter = 2 (OCCT default)
+        :type: NbIterInteger >= 2
+        :param Anisotropie = False (OCCT default)
+        :type Anisotropie: Boolean
+        :param: Tol2d = 0.00001 (OCCT default)
+        :type Tol2d: float > 0
+        :param Tol3d = 0.0001 (OCCT default)
+        :type Tol3dReal: float > 0
+        :param TolAng = 0.01 (OCCT default)
+        :type TolAngReal: float > 0
+        :param TolCurv = 0.1 (OCCT default)
+        :type TolCurvReal: float > 0
+        :param MaxDeg = 8 (OCCT default)
+        :type MaxDegInteger: Integer >= 2 (?)
+        :param MaxSegments = 9 (OCCT default)
+        :type MaxSegments: Integer >= 2 (?)
+        """
+        
+        # If thickness is 0, only a 2D surface will be returned.
+        if thickness==0:
+            combine=False
+        
+        # If a list of wires is provided, make a closed wire
+        if not isinstance(surf_edges, list): 
+            surf_edges = [o.vals()[0] for o in surf_edges.all()]
+            surf_edges = Wire.assembleEdges(surf_edges)
+            surf_edges = surf_edges.wrapped
+        
+        # Creates interpolated plate
+        def _makeplate(pnt):
+            return Solid.interpPlate(surf_edges, surf_pts, thickness, Degree, NbPtsOnCur, NbIter, Anisotropie, Tol2d, Tol3d, TolAng, TolCurv, MaxDeg, MaxSegments)
+
+        plates = self.eachpoint(_makeplate, True)
+
+        # if combination is not desired, just return the created boxes
+        if not combine:
+            return plates
+        else:
+            # combine everything --> CRASHES, probably OCC 6 CAD kernel issue, better use pushpoints through each() instead
```

Comment:
Does this also crash with clean=False?

Reply:
```python
plate_3 = cq.Workplane('XY').pushPoints(pts).interpPlate(edge_wire, surface_points, thickness, combine=True, clean=False, degree=2, nbPtsOnCur=20, nbIter=2, anisotropy=False, tol2d=0.00001, tol3d=0.0001, tolAng=0.01, tolCurv=0.1, maxDeg=8, maxSegments=9)
```
just runs forever (still waiting for the result), while
```python
def face(pos):
    return be.rotate((0,0,0),(0,0,1), 30).translate(pos).val()
plate_3 = cq.Workplane('XY').pushPoints(pts).each(face)
```
runs instantly.

Reply:
You can try reducing `tol3d` to something like 0.001 or even 0.1. Tolerance is a common culprit for me when the kernel hangs while doing an operation.

Reply:
@jmwright tried that, has no effect. I suspect the CAD kernel it does not manage to achieve a proper union of such complex shapes when CQ calls ``self.union(plates, clean=clean)`` when ``combine=True``

Reply:
@bragostin does it always crash (i.e. even for simple shapes)? Is so it would be good to disable this branch and emit a warning. Afte moving to OCCT7.4 we could revisit this.

Reply:
@adam-urbanczyk actually I realized that pushpoints is not properly managed in the ``interpPlate()`` function, I would need to work on it, maybe the crash comes from that.
Also the question of thickening inside the function or outside as a workplane method has to be decided I think.
So this PR is anyway not ready yet I would say and I agree that it is probably better to wait after moving to OCCT7.4 to proceed with it.